app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -777,53 +779,65 @@ class HomeFragment : Fragment() {
             context,
             onItemTapped = {
                 when (it) {
-                    HomeMenu.Item.Settings -> {
+                    is Item.Settings -> {
                         hideOnboardingIfNeeded()
                         nav(
                             R.id.homeFragment,
                             HomeFragmentDirections.actionGlobalSettingsFragment()
                         )
                         requireComponents.analytics.metrics.track(Event.HomeMenuSettingsItemClicked)
                     }
-                    HomeMenu.Item.SyncTabs -> {
+                    is Item.SyncedTabs -> {
                         hideOnboardingIfNeeded()
                         nav(
                             R.id.homeFragment,
                             HomeFragmentDirections.actionGlobalSyncedTabsFragment()
                         )
                     }
-                    HomeMenu.Item.Bookmarks -> {
+                    is Item.SyncAccount -> {
+                        hideOnboardingIfNeeded()
+                        val directions = if (it.signedIn) {
+                            BrowserFragmentDirections.actionGlobalAccountSettingsFragment()
+                        } else {
+                            BrowserFragmentDirections.actionGlobalTurnOnSync()
+                        }
```

Comment:
This looks similar to what we have on` BrowserToolbarMenuController#229`, we could have a helper function.
![image](https://user-images.githubusercontent.com/773158/115297763-9545a580-a12a-11eb-89f8-b9bb4995abe8.png)

Reply:
I had helper functions but @jonalmeida didn't want code cleanup in this PR so I removed them

Reply:
👍🏽

Reply:
Sorry, I'm not sure where the helper function was, my intent was to remove the `FenixAccountManager` and any refactoring code that was not related to the bug fix.

Reply:
Sorry, I just noticed code was duplicated. Agree if the code it's not related to this patch is better to not bring it in now.